ext/FerriteSparseMatrixCSR.jl

```diff
@@ -0,0 +1,80 @@
+module FerriteSparseMatrixCSR
+
+using Ferrite, SparseArrays, SparseMatricesCSR
+import Base: @propagate_inbounds
+
+@propagate_inbounds function Ferrite._assemble_inner!(K::SparseMatrixCSR, Ke::AbstractMatrix, dofs::AbstractVector, sorteddofs::AbstractVector, permutation::AbstractVector, sym::Bool)
+    current_row = 1
+    ld = length(dofs)
+    @inbounds for Krow in sorteddofs
+        maxlookups = sym ? current_row : ld
+        Kerow = permutation[current_row]
+        ci = 1 # col index pointer for the local matrix
+        Ci = 1 # col index pointer for the global matrix
+        nzr = nzrange(K, Krow)
+        while Ci <= length(nzr) && ci <= maxlookups
+            C = nzr[Ci]
+            Kcol = K.colval[C]
+            Kecol = permutation[ci]
+            val = Ke[Kerow, Kecol]
+            if Kcol == dofs[Kecol]
+                # Match: add the value (if non-zero) and advance the pointers
+                if !iszero(val)
+                    K.nzval[C] += val
+                end
+                ci += 1
+                Ci += 1
+            elseif Kcol < dofs[Kecol]
+                # No match yet: advance the global matrix row pointer
+                Ci += 1
+            else # Kcol > dofs[Kecol]
+                # No match: no entry exist in the global matrix for this row. This is
+                # allowed as long as the value which would have been inserted is zero.
+                iszero(val) || _missing_sparsity_pattern_error(Krow, Kcol)
+                # Advance the local matrix row pointer
+                ci += 1
```

Comment:
This is not covered by tests. Would it make sense to add a test with a specific field coupling (I believe that is when these conditionals become active) to check these corner cases?

Reply:
I am not sure what to do here tbh. Actually I would like to prefer to remove this case everywhere, because this is the case when the prealloacted sparsity matrix is "too small", but the values beyond its size is zero anyway. However, this would be technically a quite annoying breaking change for users running into this case.

Reply:
I meant a test like this
```julia
grid = generate_grid(Quadrilateral, (10,10));
ip = Lagrange{RefQuadrilateral,1}();
dh = DofHandler(grid); add!(dh, :u, ip); add!(dh, :v, ip); close!(dh);
coupling = [true; false;; false; true];
K = create_sparsity_pattern(SparseMatrixCSR, dh; coupling);
Ke_rand = rand(ndofs_per_cell(dh,1), ndofs_per_cell(dh,1));
Ke_zero = zeros(ndofs_per_cell(dh,1), ndofs_per_cell(dh,1));
a = start_assemble(K);
assemble!(a, celldofs(dh,1), Ke_zero); # Should work
assemble!(a, celldofs(dh,1), Ke_rand); # Should fail
```

Reply:
Is this what you had in mind?